astropy/cosmology/_src/traits/tcmb.py

```diff
@@ -0,0 +1,57 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""CMB Temperature.
+
+This is private API. See `~astropy.cosmology.parts` for public API.
+
+"""
+
+__all__ = ["TemperatureCMB"]
+
+
+from numpy.typing import ArrayLike
+
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+from astropy.units import Quantity
+
+
+class TemperatureCMB:
+    """The object has attributes and methods for computing the cosmological background temperature."""
+
+    Tcmb0: Quantity
+    """Temperature of the CMB at z=0."""
+
+    @deprecated_keywords("z", since="7.0")
+    def Tcmb(self, z: Quantity | ArrayLike) -> Quantity:
+        """Return the CMB temperature at redshift ``z``.
+
+        Parameters
+        ----------
+        z : Quantity-like ['redshift'], array-like
+            Input redshift.
+
+            .. versionchanged:: 7.0
+                Passing z as a keyword argument is deprecated.
+
+        Returns
+        -------
+        Tcmb : Quantity ['temperature']
+            The temperature of the CMB in K.
```

Comment:
```suggestion
            The temperature of the CMB.
```
First of all, if you return a `Quantity` then the unit is unambiguously a part of that `Quantity` and does not have to be written out here explicitly. Furthermore, doesn't the unit of the output depend on the unit of `Tcmb0`? Related to that, what happens if a user specifies `Tcmb0` in degrees Celsius?

Reply:
✅